src/datumaro/experimental/media.py

```diff
@@ -0,0 +1,285 @@
+                converted = img.convert("RGB")
+            elif self.format.upper() == "RGBA":
+                converted = img.convert("RGBA")
+            elif self.format.upper() == "L":
+                converted = img.convert("L")
+            else:
+                converted = img
+
+            img_array = np.array(converted, dtype=np.uint8)
+
+            # Handle BGR format by swapping R and B channels
+            if self.format.upper() == "BGR" and img_array.ndim == 3:
+                img_array = img_array[..., ::-1].copy()
+
+            # Handle channels-first format
+            if self.channels_first and img_array.ndim == 3:
+                img_array = img_array.transpose(2, 0, 1)
+
+        return img_array
+
+    @property
+    def data(self) -> np.ndarray:
+        """
+        Load and return the image data as a numpy array.
+
+        The image is loaded from disk on first access and cached in a global
+        LRU cache. Subsequent accesses return the cached data if available.
+        When the cache is full, least recently used images are automatically
+        evicted.
+
+        Returns:
+            numpy.ndarray: The image data as a numpy array with shape:
+                - (H, W, C) if channels_first is False
+                - (C, H, W) if channels_first is True
+
+        Raises:
+            FileNotFoundError: If the image file does not exist
+            PIL.UnidentifiedImageError: If the file cannot be read as an image
+        """
+        key = self._cache_key()
+
+        # Try to get from cache
+        cached = _image_cache.get(key)
+        if cached is not None:
+            return cached
+
+        # Load and cache
+        img_array = self._load_image()
+        _image_cache.put(key, img_array)
+
+        return img_array
+
+    def clear_cache(self) -> None:
+        """Remove this image from the cache to free memory."""
+        _image_cache.remove(self._cache_key())
+
+    @property
+    def width(self) -> int:
+        """Get the image width without fully loading the image data."""
+        with Image.open(self.path) as img:
+            return img.width
+
+    @property
+    def height(self) -> int:
+        """Get the image height without fully loading the image data."""
+        with Image.open(self.path) as img:
+            return img.height
+
+    @property
+    def size(self) -> tuple[int, int]:
+        """Get the image size (width, height) without fully loading the image data."""
+        with Image.open(self.path) as img:
+            return img.size
+
+    @property
+    def shape(self) -> tuple[int, ...]:
+        """
+        Get the shape of the image data.
+
+        Returns:
+            tuple: Shape of the image array. If channels_first is False: (H, W, C).
+                   If channels_first is True: (C, H, W). For grayscale: (H, W).
+        """
+        return self.data.shape
+
+    def __str__(self) -> str:
+        return str(self.path)
```

Comment:
```suggestion
    def __str__(self) -> str:
        return f"LazyImage({self.path})"
```
An object representation should contain the class name, else the user can be confused about the type.